packages/world/ts/config/v2/world.ts

```diff
@@ -39,7 +41,10 @@ export function validateWorld(world: unknown): asserts world is WorldInput {
   const scope = extendedScope(world);
   validateStore(world);
 
-  if (hasOwnKey(world, "namespaces")) {
+  if (hasOwnKey(world, "namespaces") && isObject(world.namespaces)) {
+    if (hasOwnKey(world, "namespace")) {
```

Comment:
we should check for `tables` and `systems` keys here too, otherwise runtime doesn't have the same behavior as types

Reply:
I think we also need this check in `validateWorldWithShorthands`